### DISCUSSION TITLE
Github Release - Information

### DISCUSSION THREAD

**Discussion by JcMinarro** [community-scripts/ProxmoxVE#23] [Ideas]
I know this repository doesn't release any software artifacts but would be great to have a list of changes that anyone outside of the repository can check and/or monitor.

I can see there is a `CHANGELOG.md` file with the new changes on the repository. That is great and it is partially what I mean, the problem with this file is that there isn't an easy way to be notified when new changes are added (or at least, I don't know any service to do it)
If you could add the same `CHANGELOG.md` content to the "Releases section" on Github, services such as https://newreleases.io/ can be used to be notified when new "releases" are published

**Comment by havardthom** [comment]
You could use a self-hosted Change Detection LXC to watch the changelog page https://community-scripts.github.io/Proxmox/scripts?id=Change%20Detection

Does that solve your problem?

**Reply by JcMinarro** [reply]
I didn't use "Change Detection" before, but it is an extra self-hosted service to be maintained just for something that Github already provides.

I normally use https://newreleases.io service because I centralize notifications to Gotify, but people can subscribe from github itself within "watch" section as well.
![Screenshot from 2024-11-03 10-12-20](https://github.com/user-attachments/assets/cda185a8-436b-4c53-8ba9-34440b8674e5)

I think it is a good way to notify new changes to the community

**Comment by CrazyWolf13** [comment]
I like this idea, if it's not that much extra work, this would be awesome, I love browsing my Github feed for changes and updates.

**Comment by MickLesk** [comment]
I see several problems here:

1. there are no classic releases
2. changelog.md is extended with every breaking script change or new script
3. the time problem. When should something like this be pushed? Assuming we make 5 scripts a day, is that 5 releases? Or rather once every Sunday? 

There is a lot to consider. Theoretically, it would be more for the website, with tteck his the changelog was also displayed pulsating when something was new.

**Reply by CrazyWolf13** [reply]
Thb. to me it does not depened, how I just would love to get some sort of way to get notified on updates/changes more easy, maybe generating a small rss feed via the changelog.md file would already solve this?

**Reply by MickLesk** [reply]
However, you can consider, for example, whether there is a Notify or Gotify Server that sends an update for subscribed users. (Based on Changelog) But this all has to work properly.

**Reply by havardthom** [reply]
What would be the difference between updating changelog.md vs publishing a release? The only difference I see is that a release would contain versioning and be more explicit on everything that changed compared to last release. I don't think it is a problem having several releases every day if we want

**Reply by CrazyWolf13** [reply]
@havardthom While you are right about this. considering further grow of this project, having multiple releases a day will end up in a spammy situation, which probably is probably not a good idea.

**Reply by JcMinarro** [reply]
How often is the changelog updated?
The same section the changelog contains, as far as they are markdown, can be "copy&paste" to the release version section.
A GH Action can be done to be executed at the end of the day, parsing the CHANGELOG.md file, and, if there is any change, publish a new release on GH with the new content of the CHANGELOG.

**Reply by havardthom** [reply]
Changelog has been updated 1-3 times on a single day, or more on very rare occasions https://github.com/community-scripts/ProxmoxVE/commits/main/CHANGELOG.md?before=43783920ef6e12b2efc4c0f9b9cd8ab333725c69+35

**Reply by havardthom** [reply]
I don't think spam will be an issue, we can release max 2 times on a single day for example. People do not need to be more up to speed than that tbh.

**Comment by JcMinarro** [comment]
In the meantime, I will share it here. (https://github.com/JcMinarro/ProxmoxVE-Releases/releases)
I already did a GH Action on a separate repository that is run every night, checks changes on the Changelog (Thanks @havardthom for automating Changelogs PRs ;)) and releases a new version on the "Github Releases Section" with the  CHANGELOG.md data.

With that, anyone can subscribe to new changes subscribing to GH releases, or integrating with other services like https://newreleases.io/github/JcMinarro/ProxmoxVE-Releases

**Reply by Aloe-recite** [reply]
Man, that's great! Excellent! Thank you!

**Comment by havardthom** [comment]
Fixed in https://github.com/community-scripts/ProxmoxVE/pull/462
